### PR TITLE
Move deployment collision avoidance e2e test to integration

### PR DESCRIPTION
**Pull request by janetkuo**
**What this PR does / why we need it**:

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: ref #52113

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by dims** [comment]
@janetkuo @enisoc - ok to move this out of 1.8?

File: test/integration/deployment/deployment_test.go [hunk 1]
```diff
@@ -319,3 +321,64 @@ func TestScalePausedDeployment(t *testing.T) {
...
+	}
+	_, err = tester.updateReplicaSet(newRS.Name, func(update *v1beta1.ReplicaSet) {
+		*update.Spec.Template.Spec.TerminationGracePeriodSeconds = int64(5)
+	})
+	if err != nil {
+		t.Fatalf("failed updating replicaset %s template: %v", newRS.Name, err)
+	}
+
+	// Expect deployment collision counter to increment
+	if err := wait.PollImmediate(time.Second, time.Minute, func() (bool, error) {
```
Comment (enisoc) [inline test/integration/deployment/deployment_test.go]: Based on some earlier experiments, I think we can afford to poll more frequently than 1s in integration tests. Things respond pretty quickly compared to the e2e environment. In some other integration tests I saw a 100ms interval, and moving RS/Deployment tests to that interval resulted in a significant overall speedup in my experiment.

**Review by enisoc — APPROVED** [review]
/lgtm
other than one nit

File: test/integration/deployment/deployment_test.go (hunk 1 above)
Comment (janetkuo) [reply test/integration/deployment/deployment_test.go]: @enisoc fixed in the 3rd commit, PTAL

**Pull request merged**